[CI] WriteLoadConstraintMonitorTests testHotspotCountTurnsOff failing

**Build Scans:**
- [elasticsearch-intake #32090 / part1](https://gradle-enterprise.elastic.co/s/3ia3ndoxt47hi)
- [elasticsearch-pull-request #108211 / part-1](https://gradle-enterprise.elastic.co/s/egyotpnusccuw)
- [elasticsearch-pull-request #108185 / part-1](https://gradle-enterprise.elastic.co/s/rm7us3ilcub36)
- [elasticsearch-pull-request #108085 / part-1](https://gradle-enterprise.elastic.co/s/pgir5ykmig4jo)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests.testHotspotCountTurnsOff" -Dtests.seed=B7C7B4A01F47483E -Dtests.locale=kxv-Telu-IN -Dtests.timezone=Australia/LHI -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testHotspotCountTurnsOff'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<[2, 1, 0]> but was:<[2, 1, 1]>
```

**Issue Reasons:**
- [main] 4 failures in test testHotspotCountTurnsOff (2.2% fail rate in 183 executions)
- [main] 3 failures in step part-1 (3.2% fail rate in 93 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (3.3% fail rate in 91 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -503,9 +503,6 @@ tests:
 - class: org.elasticsearch.xpack.logsdb.RandomizedRollingUpgradeIT
   method: testIndexingStandardSource
   issue: https://github.com/elastic/elasticsearch/issues/139153
-- class: org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintMonitorTests
-  method: testHotspotCountTurnsOff
-  issue: https://github.com/elastic/elasticsearch/issues/139161
 
 # Examples:
 #

--- a/server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java
@@ -660,6 +660,7 @@ public class WriteLoadConstraintMonitorTests extends ESTestCase {
         long queueLatencyThresholdMillis,
         int highUtilizationThresholdPercent
     ) {
+        assert queueLatencyThresholdMillis > 0 : "queue latency threshold must be positive";
         final Set<String> hotspotNodesSet = new HashSet<>(hotspotNodes);
         final float maxRatioForUnderUtilised = (highUtilizationThresholdPercent - 1) / 100.0f;
         ClusterInfo clusterInfo = ClusterInfo.builder()
@@ -690,7 +691,7 @@ public class WriteLoadConstraintMonitorTests extends ESTestCase {
                             new NodeUsageStatsForThreadPools.ThreadPoolUsageStats(
                                 randomNonNegativeInt(),
                                 randomFloatBetween(0f, maxRatioForUnderUtilised, true),
-                                randomLongBetween(0, queueLatencyThresholdMillis)
+                                randomLongBetween(0, queueLatencyThresholdMillis - 1)
                             )
                         )
                     );

